### PR TITLE
leftwm-theme: init at 0.1.2, leftwm-config: init at a9f2f21ece3a01d6c36610295ae3163644d3f99e

### PR DESCRIPTION
**Pull request by Denperidge**
## Description of changes
Implements leftwm-theme and leftwm-config. Based on #211955 !
Due to the original PR being about a year old, I've temporarily removed him from the maintainers meta for lightwm-theme & config. If he still has an interest in that'd be great! And if he doesn't, thanks for putting the original PR up!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/le/leftwm-config/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  meta = with lib; {
+    broken = (stdenv.isLinux && stdenv.isAarch64);
+    description = "A little satellite utility for LeftWM";
+    homepage = "https://github.com/leftwm/leftwm-config";
+    license = licenses.bsd3;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
+      denperidge
+    ];
+  };
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-config/package.nix:25]:
```suggestion
  meta = {
    broken = (stdenv.isLinux && stdenv.isAarch64);
    description = "A little satellite utility for LeftWM";
    homepage = "https://github.com/leftwm/leftwm-config";
    maintainers = with maintainers; [ denperidge ];
    license = licenses.bsd3;
    platforms = lib.platforms.linux;
  };
```

File: pkgs/by-name/le/leftwm-theme/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ stdenv, lib, fetchFromGitHub, rustPlatform, openssl, pkg-config }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "leftwm-theme";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "leftwm";
+    repo = pname;
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-theme/package.nix:9]:
```suggestion
    repo = "leftwm-theme";
```

File: pkgs/by-name/le/leftwm-theme/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    rev = "v" + version;
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-theme/package.nix:10]:
```suggestion
    rev = "refs/tags/${version};
```

File: pkgs/by-name/le/leftwm-theme/package.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    sha256 = "sha256-JCNpgdV4v0DzRmELAv+UwPAZ7O2V0KI4t8zirhhQshA";
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-theme/package.nix:11]: use `hash`

File: pkgs/by-name/le/leftwm-theme/package.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    # direct writing /tmp
+    "--skip=models::config::test::test_config_new"
+    # with network access when testing
+    "--skip=operations::update::test::test_update_repos"
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
+  OPENSSL_NO_VENDOR = 1;
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-theme/package.nix:26]: maybe `env.OPENSSL_NO_VENDOR = 1;`

File: pkgs/by-name/le/leftwm-theme/package.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  meta = with lib; {
+    broken = (stdenv.isLinux && stdenv.isAarch64);
+    description = "A theme manager for LeftWM";
+    homepage = "https://github.com/leftwm/leftwm-theme";
+    license = licenses.bsd3;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
+      denperidge
+    ];
+  };
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-theme/package.nix:37]:
```suggestion
  meta = {
    broken = (stdenv.isLinux && stdenv.isAarch64);
    description = "A theme manager for LeftWM";
    homepage = "https://github.com/leftwm/leftwm-theme";
    license = lib.licenses.bsd3;
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ denperidge ];
  };
```

File: pkgs/by-name/le/leftwm-config/0001-rm-unstable-is-some-with-feature.patch [hunk 7]
```diff
@@ -0,0 +1,93 @@
+diff --git a/src/main.rs b/src/main.rs
```
Comment (Sigmanificient) [inline pkgs/by-name/le/leftwm-config/0001-rm-unstable-is-some-with-feature.patch:1]: I dont see any `patches = [ ./0001-rm-unstable-is-some-with-feature.patch ];`. Is the patch used?

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/le/leftwm-config/0001-rm-unstable-is-some-with-feature.patch (hunk 7 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/0001-rm-unstable-is-some-with-feature.patch:1]: Oh! It was part of the original commits I've based my work upon. The patch seemed needed at the time, but I disabled it whilst keeping it as it seemed to run without, then lost track of it. Will be removed!

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -4975,6 +4975,12 @@
     githubId = 62989;
     name = "Demyan Rogozhin";
   };
+  denperidge = {
+    email = "cat@denperidge.com";
+    github = "denperidge";
+    githubId = 27348469;
+    name = "Cat";
+  };
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply maintainers/maintainer-list.nix:4983]: Will do!

File: pkgs/by-name/le/leftwm-config/package.nix (hunk 1 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-config/package.nix:25]: That's a better formatting! Will be changed

File: pkgs/by-name/le/leftwm-theme/package.nix (hunk 2 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:9]: Indeed unnecessary usage of variables, thanks for the suggestion!

File: pkgs/by-name/le/leftwm-theme/package.nix (hunk 4 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:11]: Ah, old notation mistake! On it

File: pkgs/by-name/le/leftwm-theme/package.nix (hunk 2 above)
Comment (Sigmanificient) [reply pkgs/by-name/le/leftwm-theme/package.nix:9]: - https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/le/leftwm-theme/package.nix (hunk 3 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:10]: Woopsie, that's on me!

File: pkgs/by-name/le/leftwm-theme/package.nix (hunk 5 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:26]: Assumed those were different things, seems I assumed wrong! Will be changed

File: pkgs/by-name/le/leftwm-theme/package.nix (hunk 2 above)
Comment (Denperidge) [reply pkgs/by-name/le/leftwm-theme/package.nix:9]: Woah, had no idea! Will keep a note of this, my thanks

**Comment by Denperidge** [comment]
@Sigmanificient due to not being able to retract the maintainer commit without force pushing and thus auto-closing this PR, see #317805 !

**Comment by Sigmanificient** [comment]
> @Sigmanificient due to not being able to retract the maintainer commit without force pushing and thus auto-closing this PR, see #317805 !

Not sure why force pushing would autoclose the pr :thinking:

**Comment by Sigmanificient** [comment]
I would just have kept the commit history as follows:
- `maintainers: add denperidge`
- `leftwm-theme: init at 0.1.2`
- `leftwm-config: init at 0-unstable-2024-03-23`

or if you really want to keeps yanganto's commits:
- (`leftwm-config: init at 0.1.0` from yanganto)
- (`leftwm-theme: init at unstable-2022-12-24` from yanganto)
- `maintainers: add denperidge`
- `leftwm-{config,theme}: update, migrate to by-name, refactor`

**Pull request closed without merging**